src/api/index.js

```diff
@@ -343,6 +358,82 @@ export function getInstallProperties() {
   return get(uri);
 }
 
+
+export async function patchServiceAccount(
+  serviceAccountName,
+  namespace,
+  secretName
+) {
+  const uri = getKubeAPI('serviceaccounts', {
+    name: serviceAccountName,
+    namespace
+  });
+  const patch1 = await patchAddSecret(uri, secretName);
+  return patch1;
+}
+
+// Get list of service accounts where secrets patched to, returns service account in full
+export async function getSecretServiceAccountList(saList, secretName) {
+  const secretServiceAccountList = [];
+
+  if (saList.length > 0) {
+    saList.forEach(element => {
+      const saSecretList = element.secrets;
+      let secretFound = false;
+      saSecretList.forEach(element1 => {
+        if (element1.name === secretName) {
+          secretFound = true;
+        }
+      });
+      if (secretFound === true) {
+        secretServiceAccountList.push(element);
+      }
+    });
+  }
+
+  return secretServiceAccountList;
+}
+
+export function getIndexOfSecret(sa, secretName) {
+  let x = 0;
+
+  let indexOfSecret = -1;
+  sa.forEach(element => {
+    if (element.name === secretName) {
+      indexOfSecret = x;
+    }
+    x += 1;
+  });
+
+  return indexOfSecret;
+}
+
+export async function getIndexAndRemove(sa, secretName, namespace) {
+  const indexOfSecret = await getIndexOfSecret(sa.secrets, secretName);
+
+  // Should never be -1 as means the secret is not found
+  if (indexOfSecret === -1) {
+    return 'Not Found';
```

Comment:
Should this instead throw an error which we can catch and do stuff with later? Instead of returning a string

Reply:
Done 😸